packages/in_app_purchase/ios/Classes/InAppPurchasePlugin.m

```diff
@@ -204,24 +204,25 @@ - (void)finishTransaction:(FlutterMethodCall *)call result:(FlutterResult)result
     return;
   }
   NSString *identifier = call.arguments;
-  SKPaymentTransaction *transaction =
-      [self.paymentQueueHandler.transactions objectForKey:identifier];
-  if (!transaction) {
+  NSMutableArray *transactions = [self.paymentQueueHandler.transactions objectForKey:identifier];
+  if (!transactions) {
     result([FlutterError
         errorWithCode:@"storekit_platform_invalid_transaction"
               message:[NSString
                           stringWithFormat:@"The transaction with transactionIdentifer:%@ does not "
                                            @"exist. Note that if the transactionState is "
                                            @"purchasing, the transactionIdentifier will be "
                                            @"nil(null).",
-                                           transaction.transactionIdentifier]
+                                           identifier]
               details:call.arguments]);
     return;
   }
   @try {
+    for (SKPaymentTransaction *transaction in transactions) {
+      [self.paymentQueueHandler finishTransaction:transaction];
+    }
```

Comment:
Is it ok to abort the loop if an exception is thrown or should you move try/catch inside the loop.

Reply:
Is there a way for the user to know which transactions are failed when finishing?